src/java.desktop/windows/native/libawt/windows/awt_Toolkit.h

```diff
@@ -475,7 +475,7 @@ class AwtToolkit {
     BOOL m_verifyComponents;
     BOOL m_breakOnError;
 
-    BOOL  m_breakMessageLoop;
+    volatile BOOL  m_breakMessageLoop;
```

Comment:
Does this volatile modifier resolve the now-removed infinite loop in `while (!tk.IsDisposed())` in `WToolkit_shutdown`?

Reply:
> Does this volatile modifier resolve the now-removed infinite loop in `while (!tk.IsDisposed())` in `WToolkit_shutdown`?

The loop should not be removed.

Reply:
Unfortunately, volatile modifier has no effect if infinite loop is reinstated..

Reply:
> > Does this volatile modifier resolve the now-removed infinite loop in `while (!tk.IsDisposed())` in `WToolkit_shutdown`?
> 
> The loop should not be removed.

No, it should not, as you noted previously.

However, making `m_breakMessageLoop` volatile does not look right either. If `QuitMessageLoop` is called from `!IsMainThread()` thread, it is posted as a message to run on the correct thread. Thus `m_breakMessageLoop` should be accessed on a single thread only; if it's the case, volatile is unneeded.

And @prsadhuk's latest test confirms it. There must be something else…

Reply:
I ran the test locally for several iterations but it does not fail locally so I believe it's a test issue. Re reverting the test changes made in 1st webrev...

Reply:
I do not see a reason to change the test until the root cause of the hang is not recognized.

Reply:
There were many instances faced during test sprint where a test was failing in one platform and we added the stability code of waitForIdle,delay etc to make it work. It was not cited as product bug...I believe it's the same here...I am not able to reproduce it locally at all, and I also checked with other who was also not able to reproduce it locally. It  only failed in mach5 and it passed once the test is modified with stability fixes...I dont see a reason to stall it.

Reply:
No this is not the same, this a hang in the product due to failed toolkit shutdown, please create one more bug and attach the old test to it reproduce the bug.

Reply:
Ok. JDK-8261016